Game/marvin.cpp

```diff
@@ -6,132 +6,130 @@
 #include "world/objects/npc.h"
 #include "camera.h"
 #include "gothic.h"
+#include "utils/stringutil.h"
+
+Marvin::Marvin(Gothic& gothic)
+: gothic( gothic )
+, commandHandlers()
+{
+  commandHandlers["cheat full"] = CommandFunction(&Marvin::handleCheatFull);
+  commandHandlers["camera autoswitch"] = CommandFunction(&Marvin::handleCameraAutoSwitch);
+  commandHandlers["camera mode"] = CommandFunction(&Marvin::handleCameraMode);
+  commandHandlers["toggle camdebug"] = CommandFunction(&Marvin::handleToggleCamDebug);
+  commandHandlers["toggle camera"] = CommandFunction(&Marvin::handleToggleCamera);
+  commandHandlers["insert"] = CommandFunction(&Marvin::handleInsert);
+  }
 
-Marvin::Marvin() {
-  cmd = std::vector<Cmd>{
-    {"cheat full",        C_CheatFull},
+void Marvin::autoComplete(std::string& inputString) {
```

Comment:
This code do very-much same as `Marvin::recognize` use to preform, but in veery expensive manner.
OK, this path is not time-critical, but why downgrade algorithm?

Reply:
Truth be told, I did not really understand what the recognize method really does.
And because I was already changing the the way the data is stored I just replaced the code.

This was also not the first version I implemented that day, the first implementation was much more clean but did not have the same result as your original implementation.

I will look into the recognize function once more and try to understand exactly whats happening and maybe I can re-use it.
 I was not sure if it can handle commands with parameters like the "insert" command?
Where only the command can be recognized, but not the rest of the string.